src/main/java/slimeknights/tconstruct/plugin/crt/CRTHelper.java

```diff
@@ -0,0 +1,150 @@
+import com.google.gson.JsonElement;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.Color;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.registries.ForgeRegistries;
+import slimeknights.tconstruct.TConstruct;
+import slimeknights.tconstruct.library.MaterialRegistry;
+import slimeknights.tconstruct.library.TinkerRegistries;
+import slimeknights.tconstruct.library.materials.IMaterial;
+import slimeknights.tconstruct.library.materials.Material;
+import slimeknights.tconstruct.library.materials.MaterialId;
+import slimeknights.tconstruct.library.modifiers.Modifier;
+import slimeknights.tconstruct.library.modifiers.ModifierEntry;
+import slimeknights.tconstruct.library.modifiers.ModifierId;
+import slimeknights.tconstruct.library.tinkering.IMaterialItem;
+import slimeknights.tconstruct.library.tools.item.ToolCore;
+
+import java.util.stream.Collectors;
+
+public class CRTHelper {
+  
+  public static MaterialId getMaterialId(String materialId) {
+    MaterialId material = MaterialId.tryCreate(materialId);
+    if(material == null) {
+      throw new IllegalArgumentException("Invalid ResourceLocation provided! Provided: " + materialId);
+    }
+    IMaterial foundMaterial = MaterialRegistry.getMaterial(material);
+    if(foundMaterial == Material.UNKNOWN) {
+      throw new IllegalArgumentException("Material does not exist! Provided: " + materialId);
+    }
+    
+    return material;
+  }
+  
+  public static Modifier getModifier(String name) {
+    ModifierId resultId = ModifierId.tryCreate(name);
+    if(resultId == null) {
+      throw new IllegalArgumentException("Invalid ResourceLocation provided! Provided: " + name);
+    }
+    Modifier resultModifier = TinkerRegistries.MODIFIERS.getValue(resultId);
+    if(resultModifier == null) {
+      throw new IllegalArgumentException("Modifier does not exist! Provided: " + resultId);
+    }
+    return resultModifier;
+  }
+  
+  @SubscribeEvent
+  public void onExampleCollection(ExampleCollectionEvent event) {
+    event.addResource(location("alloying"));
+    event.addResource(location("beheading"));
+    event.addResource(location("casting_basin"));
+    event.addResource(location("casting_table"));
+    event.addResource(location("entity_melting"));
+    event.addResource(location("fuel"));
+    event.addResource(location("material"));
+    event.addResource(location("melting"));
+    event.addResource(location("molding_basin"));
+    event.addResource(location("molding_table"));
+    event.addResource(location("part_builder"));
+    event.addResource(location("tinker_station"));
+  }
+  
+  private ResourceLocation location(String location) {
+    return new ResourceLocation(TConstruct.modID, TConstruct.modID + "/" + location);
```

Comment:
I assume this is an API detail, but any reason the mod ID is here twice?

Reply:
So the first mod ID is so that it can find the files in the first place, the second is the folder that the files go in when someone runs `/ct examples`. So with how it is setup now, running `/ct examples` produces the following:
![image](https://user-images.githubusercontent.com/5553605/114685064-d3e5f500-9d11-11eb-876a-605b202a0879.png)

If we removed the second modid, it would be with the rest of the CraftTweaker examples.

I can change it to be something else if you would like? maybe "tinkers_construct"?

Reply:
Nah, was just curious as it looked unintentional. Maybe add a comment explaining the two mod IDs?

Reply:
The event is now deprecated, so no need for the method anymore.